app/models/report.rb

```diff
@@ -93,6 +93,208 @@ def cleanup_report
     end
   end
 
+  def self.generate_whole_project_report(project, current_user, current_team)
+    report_contents = []
+    report_contents << {
+      'type_of' => 'project_header',
+      'id' => {
+        'project_id' => project.id
+      },
+      'sort_order' => nil,
+      'children' => []
+    }
```

Comment:
This hash repeats through the whole code, it could be refactored with a helper method containing ID and type_of.